metadata_request.go

```diff
@@ -45,6 +45,32 @@ func NewMetadataRequest(version KafkaVersion, topics []string) *MetadataRequest
 	return m
 }
 
+func NewNegotiatedMetadataRequest(version KafkaVersion, brokerMinMaxVersions [2]int16, topics []string) (*MetadataRequest, error) {
```

Comment:
I would recommend using a `minVersion, maxVersion int16` for ease of understanding which of the two `int16`s are which.

Reply:
Updated